packages/contract-notifier/src/ContractNotifier.js

```diff
@@ -0,0 +1,119 @@
+   * @param {Object} networker Used to send the API requests.
+   * @param {Function} getTime Returns the current time.
+   * @param {String} apiEndpoint API endpoint to monitor.
+   * @param {Integer} maxTimeTillExpiration Period in seconds to look for upcoming contract expirations.
+   */
+  constructor({ logger, networker, getTime, apiEndpoint, maxTimeTillExpiration }) {
+    this.logger = logger;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.apiEndpoint = apiEndpoint;
+    this.maxTimeTillExpiration = maxTimeTillExpiration;
+  }
+
+  async checkUpcomingExpirations() {
+    this.logger.debug({
+      at: "ExpirationsNotifier",
+      message: "Checking for upcoming expirations",
+      apiEndpoint: this.apiEndpoint,
+      maxTimeTillExpiration: this.maxTimeTillExpiration,
+    });
+
+    const currentTime = await this.getTime();
+    const notifiedExpirations = await this.getNotifiedExpirations();
+    const expirationPeriod =
+      this.maxTimeTillExpiration > 259200
+        ? parseInt(this.maxTimeTillExpiration / 86400) + " days"
+        : parseInt(this.maxTimeTillExpiration / 3600) + " hours";
+    const apiUrl = this.apiEndpoint + "/global/listActive";
+    const activeContracts = await this.networker.getJson(apiUrl, { method: "post" });
+    const expiringContracts = activeContracts
+      .map((contract) => {
+        if (!contract.type || !contract.expirationTimestamp) {
+          return null;
+        }
+        let tokenName;
+        if (contract.type === "emp") {
+          tokenName = contract.tokenName;
+        } else if (contract.type === "lsp") {
+          tokenName = contract.longTokenName;
+        } else {
+          tokenName = "";
+        }
+        const expirationUtcString = new Date(contract.expirationTimestamp * 1000).toUTCString();
+        return {
+          // UMA API currently supports only Ethereum mainnet, thus chainId of 1 is hardcoded here:
+          chainId: 1,
+          address: contract.address,
+          expirationTimestamp: contract.expirationTimestamp,
+          tokenName: tokenName,
+          expirationUtcString: expirationUtcString,
+        };
+      })
+      .filter((contract) => {
+        return (
+          contract &&
+          contract.expirationTimestamp - currentTime <= this.maxTimeTillExpiration &&
+          contract.expirationTimestamp > currentTime &&
+          (!Object.keys(notifiedExpirations).includes(contract.chainId + "_" + contract.address) ||
+            contract.expirationTimestamp -
+              notifiedExpirations[contract.chainId + "_" + contract.address].notificationTimestamp >
+              this.maxTimeTillExpiration)
+        );
+      });
+
+    let mrkdwn = `Following contracts are expiring in ${expirationPeriod}:`;
+    for (let contract of expiringContracts) {
+      mrkdwn =
+        mrkdwn +
+        `\n- ${createEtherscanLinkMarkdown(contract.address, contract.chainId)}:` +
```

Comment:
in future we can add the chainId on this, when going multi-chain. also, if the API has any pricing information on the token (TVL or whatever) that could be useful.

Reply:
One thing I do think you should add is the liveness that the expiration price request will have

Reply:
that would be good, but UMA APIs currently do not expose the OO liveness